### PR TITLE
Use clipped config for asset config, allow a video size vs asset size

### LINKED ISSUE #950: Remove application/zip file types
This is old and should probably be hit with some jenga.

https://github.com/pgharts/trusty-cms/blob/master/app/models/asset.rb#L4

### LINKED ISSUE #977: Deprecate old video file types
We should only support the following:
video/mp4
video/webm
video/quicktime
video/mpeg
video/ogg
video/mp2t
application/x-mp4

vendor/extensions/clipped-extension/clipped_extension.rb
 AssetType.new :video

### LINKED ISSUE #978: Deprecate Doc Asset types
We are not using these. Let's remove them entirely. We tend to use just PDFs
vendor/extensions/clipped-extension/clipped_extension.rb

Just deprecate these entirely:  AssetType.new :document

### LINKED ISSUE #979: Clean up audio file types
vendor/extensions/clipped-extension/clipped_extension.rb

We have several outmoded audio file types and update the rest. Let's only support these:
audio/mpeg
audio/ogg
audio/wav
audio/flac
audio/mp4

### LINKED ISSUE #982: Video Upload Errors
Not sure when this broke, but video upload is erroring,

To repro:
1) Assets tab
2) Upload Asset
3) Select an mp4 under 5MB

500 occurs.

### PR DESCRIPTION
**Pull request by danigirl329**
This pull request refactors and improves asset upload validation in the application, especially around file type and file size checks. It replaces the previous static list of approved content types with a more flexible system based on `AssetType`, introduces separate file size limits for videos and other assets, and enhances error handling and messaging. The changes also update tests to cover new validation logic.

**Validation and configuration improvements:**

* Replaces the static `APPROVED_CONTENT_TYPES` constant in `Asset` with a dynamic method that pulls approved MIME types from `AssetType`, making it easier to manage supported file types. (`app/models/asset.rb`, `app/controllers/admin/assets_controller.rb`, [[1]](diffhunk://#diff-0545770a3fbf7606692cd849fe6ca5dd4b454b9a203b8588657abbe7f0ea30b8L4-L5) [[2]](diffhunk://#diff-52d98c35c625099704543c387ac70e728efb04bfa7125d73bd81109907ae451fL4) [[3]](diffhunk://#diff-52d98c35c625099704543c387ac70e728efb04bfa7125d73bd81109907ae451fL86-R88)
* Adds separate maximum file size settings for general assets and for videos, configurable via `TrustyCms.config`, and updates validation logic accordingly. (`app/models/asset.rb`, `config/initializers/trusty_cms_config.rb`, [[1]](diffhunk://#diff-0545770a3fbf7606692cd849fe6ca5dd4b454b9a203b8588657abbe7f0ea30b8R38-R46) [[2]](diffhunk://#diff-c6779c1298604280a21c36ff135b0ba491f24a3418d444db1a724cc67ff59785L32-R33)
* Implements a custom validation method `approved_content_type` to ensure only files with approved MIME types are accepted. (`app/models/asset.rb`, [app/models/asset.rbR237-R246](diffhunk://#diff-0545770a3fbf7606692cd849fe6ca5dd4b454b9a203b8588657abbe7f0ea30b8R237-R246))

**Error handling and user feedback:**

* Improves error handling in the asset upload controller, ensuring that users are redirected with appropriate error messages when uploads fail due to validation errors. (`app/controllers/admin/assets_controller.rb`, [app/controllers/admin/assets_controller.rbR67-R75](diffhunk://#diff-52d98c35c625099704543c387ac70e728efb04bfa7125d73bd81109907ae451fR67-R75))
* Updates localization strings to provide clearer feedback for unsupported file types and file size errors. (`config/locales/en.yml`, [config/locales/en.ymlR20-R22](diffhunk://#diff-44438ce218f5287c58d0017f965d888715635d94280669896f75841fbd7b4cd7R20-R22))

**Testing:**

* Expands model tests to cover the new validation logic, including checks for approved content types, maximum asset size, and maximum video size, as well as correct handling of edge cases. (`spec/models/asset_spec.rb`, [spec/models/asset_spec.rbR14-R93](diffhunk://#diff-b87e7a968ce874cb0f75453b3a3ddad13a7341ba4a70bd8356f9782150584fe0R14-R93))

### CODE REVIEW

File: app/controllers/admin/assets_controller.rb [hunk 1]
```diff
@@ -64,13 +64,15 @@ def create
...
     end
 
     if asset_params[:for_attachment]
       render partial: 'admin/page_attachments/attachment', collection: @page_attachments
-    else
-      response_for :create
+    elsif @error.present?
+      flash[:error] = @error
+      redirect_to new_admin_asset_path
     end
```
[the remark this answers is not in the document]
Comment (danigirl329) [reply app/controllers/admin/assets_controller.rb:78]: resolved

File: spec/models/asset_spec.rb [hunk 2]
```diff
@@ -11,30 +11,86 @@ def upload_fixture(filename, content_type)
...
+        tempfile.binmode
+        tempfile.write('0' * (max_size.megabytes + 1))
+        tempfile.rewind
+
+        uploaded = Rack::Test::UploadedFile.new(tempfile.path, 'video/mp4')
+        asset = described_class.new(asset: uploaded)
+
+        expect(asset).not_to be_valid
+        expect(asset.errors[:asset]).to include(a_string_matching(/#{max_size} MB/))
+      end
```
[the remark this answers is not in the document]
Comment (danigirl329) [reply spec/models/asset_spec.rb]: Fixed.

File: app/models/asset.rb [hunk 3]
```diff
@@ -37,14 +35,15 @@ class Asset < ActiveRecord::Base
...
-            presence: true,
-            blob:
-              {
-                content_type: APPROVED_CONTENT_TYPES,
-                size_range: 1..10.megabytes,
-              }
+  validates :asset, presence: true
+  validates :asset, blob: { size_range: 1..TrustyCms.config['assets.max_asset_size'].to_i.megabytes }, if: -> { asset.attached? && !video_content_type? }
+  validates :asset, blob: { size_range: 1..TrustyCms.config['assets.max_video_size'].to_i.megabytes }, if: -> { asset.attached? && video_content_type? }
+  validate :approved_content_type, if: -> { asset.attached? }
```
[the remark this answers is not in the document]
Comment (danigirl329) [reply app/models/asset.rb:45]: This was refactored here: https://github.com/pgharts/trusty-cms/commit/5d6e974c7437abde552c97c7f8a3b80d1b6eabb9

**Review by greg-suehr — APPROVED** [review]
I tested a changing the Max Video Size config and uploading in number of formats. No issues with these change, but some suprises worth mentioning:

1. Uploading an `.avif` file type succeeded, but crashed when generating a thumbnail for `/admin/assets`. I had to update my local `imagemagick` to get an AVIF codec from `libheif`.
2. Uploading multiple files allows supported file types to quietly succeed while an error will display if any file types are unsupported (and they won't be uploaded). Just ambiguous UI we could modify to either disallow multiple uploads or make it clear the state of each.


Things that work:

File Types That Are Now Be Accepted
- [x] Uploading a .webp image.
- [x] Uploading an .avif image.
- [x] Uploading an .mp4 video.
- [x] Uploading a .webm video.
- [x] Uploading an .flac audio file.
- [x] Uploading a .wav audio file.

File Types That Are Now Rejected
- [x] Try uploading a .doc (Word) file.
- [x] Try uploading a .xls (Excel) file.
- [x] Try uploading a .txt plain text file.

Video Size Limit
- [x] Go to Setting > General and confirm the new "Max Video Size" field, default value of 50 MB.
- [x]  "Max Video Size" to 10 MB and "Max Asset Size" to 1 MB.
- [x] Try uploading a video file larger than 10 MB, rejected with an error mentioning “10 MB".
- [x] Try uploading a video file smaller than 2 MB but larger than 1 MB
- [x] Try uploading a non-video file (e.g., a PNG) larger than 1 MB, rejected with an error mentioning "1 MB".
- [x] Try uploading a non-video file smaller than 1 MB. 

Zero Config Edge Case
- [x] Temporarily remove or blank out the assets.max_asset_size config value in the database, displays an error message mentioning “0 MB”.

Section 5: Multiple File Upload Errors
- [x] Find a way to upload multiple files at once
- [x] Upload a batch where at least one files have unsupported types (one quietly succeeds, other fails)

Existing Assets
- [x] Open an existing asset record that was previously uploaded, edit and resave.

Regression 
- [x] Upload a standard PNG image.
- [x] Upload a PDF.
- [x] Upload an MP3 audio file.
- [x] Attach an asset to a page and confirm it renders correctly on the front end.
